tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py

```diff
@@ -149,23 +158,31 @@ def _clean_log_config(log):
         desired_keys = ["file_path", "log_stream_name"]
         return {key: log[key] for key in desired_keys}
 
-    def _get_relevant_logs(self):
-        """Get subset of all log configs that apply to this cluster's scheduler/os combo."""
-        # Figure out which logs are relevant to the master and computes for this cluster
-        logs = self._read_log_configs_from_master()
+    def _filter_logs(self, logs):
+        """Populate self._relevant_logs with logs appropriate for the two different node types."""
+        # Batch doesn't really have compute nodes in the traditional sense.
+        desired_node_roles = {"MasterServer"} if self.scheduler == "awsbatch" else {"MasterServer", "ComputeFleet"}
         for log in logs:
             if self.scheduler not in log.get("schedulers") or self.platform not in log.get("platforms"):
                 continue
-            for node_role in log.get("node_roles"):
+            for node_role in set(log.get("node_roles")) & desired_node_roles:
```

Comment:
Minor: is the `ComputeFleet` node_role in `cloudwatch_log_files.json` used when running a cluster with awsbatch? If not should we remove the role?

Reply:
On this line, `log` represents a JSON object with this format:
```
{
      "timestamp_format_key": "default",
      "file_path": "/var/log/cfn-init-cmd.log",
      "log_stream_name": "cfn-init-cmd",
      "schedulers": [
        "awsbatch",
        "sge",
        "slurm",
        "torque"
      ],
      "platforms": [
        "amazon",
        "centos",
        "ubuntu"
      ],
      "node_roles": [
        "ComputeFleet",
        "MasterServer"
      ]
}
```

`desired_node_roles` is a set that only contains `MasterServer` when batch is the scheduler. I think that has the effect that you're getting at with your question.

Removing it from the log entry in the data file itself wouldn't make sense because it is a log that would be present on compute nodes for traditional schedulers.

Reply:
I would suggest to add a comment like, "ComputeFleet role is defined in log config for awsbatch, currently not used, hence skipping tests for ComputeFleet role when running with awsbatch"